src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90

```diff
@@ -1128,6 +1153,123 @@ subroutine calc_slope_functions_using_just_e(h, G, GV, US, CS, e)
 
 end subroutine calc_slope_functions_using_just_e
 
+!! Computes UH_grad and VH_grad for gradient model (Khani & Dawson, JAMES 2023)
+subroutine calc_slope_functions_with_gradient_model(h, G, GV, US, CS, e, uh, vh)
+  type(ocean_grid_type),                       intent(inout) :: G  !< Ocean grid structure
+  type(verticalGrid_type),                     intent(in)    :: GV !< Vertical grid structure
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)),   intent(inout) :: h  !< Layer thickness [H ~> m or kg m-2]
+  real, dimension(SZIB_(G),SZJ_(G),SZK_(GV)),  intent(in) :: uh    !< Interface height times u [ZU ~> m2 s-1]
+  real, dimension(SZI_(G),SZJB_(G),SZK_(GV)),  intent(in) :: vh    !< Interface height times v [ZU ~> m2 s-1]
+  type(unit_scale_type),                       intent(in)    :: US !< A dimensional unit scaling type
+  type(VarMix_CS),                             intent(inout) :: CS !< Variable mixing control structure
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)+1), intent(in)    :: e  !< Interface position [Z ~> m]
+  ! type(thermo_var_ptrs),                     intent(in)    :: tv !< Thermodynamic variables
+  ! Local variables
+  !real :: dz_tot(SZI_(G),SZJ_(G)) ! The total thickness of the water columns [Z ~> m]
+  ! real :: dz(SZI_(G),SZJ_(G),SZK_(GV)) ! The vertical distance across each layer [Z ~> m]
+  real :: Ux_Hx(SZIB_(G), SZJ_(G))  ! X-slope of U and H  [T-1 ~> s-1]
+  real :: Uy_Hy(SZI_(G), SZJB_(G))  ! Y-slope of U and H  [T-1 ~> s-1]
+  real :: Vx_Hx(SZIB_(G), SZJ_(G))  ! X-slope of V and H  [T-1 ~> s-1]
+  real :: Vy_Hy(SZI_(G), SZJB_(G))  ! Y-slope of V and H  [T-1 ~> s-1]
+  real :: H_cutoff      ! Local estimate of a minimum thickness for masking [H ~> m or kg m-2]
+  !real :: dZ_cutoff     ! A minimum water column depth for masking [H ~> m or kg m-2]
+  real :: h_neglect     ! A thickness that is so small it is usually lost
+                        ! in roundoff and can be neglected [H ~> m or kg m-2].
+  real :: graduh        ! Gradient model frequency, zonal transport [T-1 ~> s-1]
```

Comment:
As the code is currently written `graduh` has units of `[H L-1 T-1 ~> s-1 or kg m-3 s-1]`.